html-templates/projects/projectEdit.tpl

```diff
@@ -56,11 +56,11 @@
                         <input name="ChatChannel" id="field-chat-channel" class="form-control" placeholder="train_schedule_analyzer" value="{refill field=ChatChannel default=$Project->ChatChannel}" pattern="[A-Za-z0-9_-]+" title="Hash tag containing only letters, numbers, dashes or underscores without leading #"/>
                     </div>
 
-<!--                     <select class="form-control select2-tags" multiple="multiple">
-                      <option selected="selected">orange</option>
-                      <option>white</option>
-                      <option>purple</option>
-                    </select> -->
+                    <select class="form-control select2-tags" multiple="multiple">
+                        {foreach item=Topic from=Tag::getTagsString($Project->Tags, topic)}
```

Comment:
@themightychris What should the value for `from` be here? This doesn't return any options/topics

Reply:
@marcelkooi that value looks correct.

To debug, you could use [dwoo's dump plugin](http://dwoo.org/plugins/functions/dump.html) to see what values are being worked with, for example put `{dump $Project->Tags}` just outside of the `<select>` markup.

[`Tag::getTagsString`](https://github.com/JarvusInnovations/emergence-skeleton/blob/develop/php-classes/Tag.class.php#L401) is a helper method that should return a comma-delimited list of tags, given an array of tag objects. It's optional second parameter is a "tag prefix" to filter the list down to and strip off from the output.

Behind-the-scenes, laddr uses tag prefixes to maintain different sets of tag namespaces within the global tagging system. If you take a look for example at [the data for electME](https://codeforphilly.org/projects/electme?format=json&include=Tags) you'll see that a number of the tags it has have the `topic.` prefix. You can see what this results in if I make the call on the PHP shell for the live instance:

```console
php > $Project = Laddr\Project::getByHandle('electme');
php > var_export(Tag::getTagsString($Project->Tags, 'topic'));
'Voting, Elections, Elected Positions, Philadelphia, Party Politics, Democracy, Democratic Party, Republican Party'
php >
```

I have a feeling you're just looking at a project that doesn't have any topic tags yet

Reply:
That makes sense -- I was thinking there was a list of tags that users could choose from but now I realize they can enter anything.

One issue with using select2 here is that I'm not seeing an obvious way to extract the `value` from the select list. You can enter in whatever tag you want but it won't be saved to the db. We could add extra plumbing and create a hidden input field but I'm not sure if that's the right path. Did you have an idea of what to do here?